nikola/plugin_categories.py

```diff
@@ -329,16 +328,8 @@ def split_metadata(self, data):
         This splits in the first empty line that is NOT at the beginning
         of the document, or after YAML/TOML metadata without an empty line.
         """
-        if data.startswith('---'):  # YAML metadata
-            split_result = re.split('(\n---\n|\r\n---\r\n)', data.lstrip(), maxsplit=1)
-        elif data.startswith('+++'):  # TOML metadata
-            split_result = re.split('(\n\\+\\+\\+\n|\r\n\\+\\+\\+\r\n)', data.lstrip(), maxsplit=1)
-        else:
-            split_result = re.split('(\n\n|\r\n\r\n)', data.lstrip(), maxsplit=1)
-        if len(split_result) == 1:
-            return '', split_result[0]
-        # ['metadata', '\n\n', 'post content']
-        return split_result[0], split_result[-1]
+        meta, content, _ = split_metadata(data)
```

Comment:
I don't understand this one. Is codacy maybe confusing the imported `utils.split_metadata` with `PageCompiler.split_metadata`?